utils.py

```diff
@@ -363,6 +382,55 @@ def add_cmd(self, func, name=None, featured=False, aliases=None):
         self.commands[name].append(func)
         return func
 
+    def call_ctcp(self, irc, source, cmd, args):
+        """
+        Calls a PyLink bot CTCP handler. source is the caller's UID, cmd is
+        the CTCP command, and args is a list of space-separated arguments.
+        """
+        irc.called_in = source
+        irc.called_by = source
+    
+        funcs = None
+        need_global = False
+        if cmd in self.ctcp_commands:
+            # Try our own handler first
+            funcs = self.ctcp_commands[cmd]
+        elif cmd in world.services['pylink'].ctcp_commands:
+            # Fall back to global handler on PyLink service
+            funcs = world.services['pylink'].ctcp_commands[cmd]
+            need_global = True
+        else:
+            log.info('(%s/%s) Received unknown CTCP command %r from %s', irc.name, self.name, cmd, irc.get_hostmask(source))
+            return
+                
+        # Run handlers
+        log.info('(%s/%s) Handling CTCP %r for %s', irc.name, self.name, cmd, irc.get_hostmask(source))
+        for func in funcs:
```

Comment:
Per above, I think CTCP listeners should be singular (so this iteration isn't needed).